[김광일] Mission 2: 모델링

Kimgwangil opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- models.py 수정
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- class User 생성
- name, email, password, phone_number  필드 생성

---
- class Username 변경
- name -> first_name / last_name으로 분기
- email 필드 unique 상태 추가 ( 유저의 이메일 중복 저장을 방지)
- password 길이 증가 ( 100 ) -> 비밀번호는 후에 토큰으로 받기 떄문에
- create_at / updated_at
---
class name / db_table 재변경


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- `create_at` 과 `updated_at` 을 추가해주는 이유?
   - 가입 날짜와 업데이트 날짜를 받아서 후에 오래된 회원을 관리하거나, 중복 가입을 방지하기 위해서 일까?

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Kimgwangil committed:
Add : models added class_user

kyeongraekim commented on students/gwangil/users/models.py:
```diff
@@ -0,0 +1,11 @@
+
+# Create your models here.
+class User(models.Model):
+    name         = models.CharField(max_length=10)
```
id(흔히 한국인들이 사용하는 계정의 명칭) → username
name → first_name, last_name

위와 같이 기획해 보는 것은 어떨까요?

kyeongraekim commented on students/gwangil/users/models.py:
```diff
@@ -0,0 +1,11 @@
+class User(models.Model):
+    name         = models.CharField(max_length=10)
+    email        = models.CharField(max_length=50)
+    password     = models.CharField(max_length=20)
```
password property는 나중에 해시(암호화)된 비밀번호 길이를 고려해서 넉넉하게 100 이상으로 설정해주세요.

kyeongraekim commented on students/gwangil/users/models.py:
```diff
@@ -0,0 +1,11 @@
+# Create your models here.
+class User(models.Model):
+    name         = models.CharField(max_length=10)
+    email        = models.CharField(max_length=50)
```
email field에 unique 속성을 추가하셔서 유저가 중복 저장되는 문제가 발생하지 않도록 수정해주세요!
공식 문서에서 그 방법을 찾아보고 적용해주세요.

kyeongraekim reviewed (changes requested):
아래 내용 확인하여서 수정해주세요!~

kyeongraekim commented on students/gwangil/users/models.py:
```diff
@@ -0,0 +1,11 @@
+    email        = models.CharField(max_length=50)
+    password     = models.CharField(max_length=20)
+    phone_number = models.CharField(max_length=20)
+
```
사용자/회원 데이터의 이력 관리를 위해서 created_at, updated_at column을 추가해주세요.

created_at = models.DateTimeField(auto_now_add=True)
updated_at = models.DateTimeField(auto_now=True)

Kimgwangil committed:
Fix : fixed models.py
class User -> class Username
name -> first_name / last_name
email -> added unique
password -> extended max_length
db_table = users -> usernames
Add :
created_at / updated_at

kyeongraekim commented on students/gwangil/users/models.py:
```diff
@@ -0,0 +1,14 @@
+from django.db import models
+
+# Create your models here.
+class Username(models.Model):
```
해당 클래스 명이 Username이면 너무 국한 적인 내용만 포함시킬 것 같습니다. 하위 내용이 포괄적으로 담겨도 의미적으로 문제가 없게 User로 바꾸어주세요. 그에 맞게 디비 테이블명도 수정해주세요!~

kyeongraekim reviewed (changes requested):
아래 내용 참고하여 추가 수정 해주세요!~

Kimgwangil committed:
Fix : class name
Username -> User
db_table
usernames -> users

kyeongraekim reviewed (approved):
LGTM!~
